flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -751,6 +760,8 @@ public void testPartitionTypesInBatchMode() {
         assertThat(
                 verticesSorted.get(4) /* forward - sink */,
                 hasOutputPartitionType(ResultPartitionType.PIPELINED_BOUNDED));
+
+        assertEquals(JobType.BATCH, jobGraph.getJobType());
```

Comment:
A separate test similar to the STREAMING case would be good. Is there maybe a test somewhere for `StreamGraphGenerator#shouldExecuteInBatchMode` that we could extend?

Otherwise, I'd prefer a test like this:

```
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setRuntimeMode(RuntimeExecutionMode.BATCH);
        env.fromElements("test").addSink(new DiscardingSink<>());
        JobGraph jobGraph = StreamingJobGraphGenerator.createJobGraph(env.getStreamGraph());
        assertEquals(JobType.BATCH, jobGraph.getJobType());
```

Reply:
I couldn't find any test for `shouldExecuteInBatchMode()`. I'm adding a separate test.